src/main/character/Character.java

```diff
@@ -1,5 +1,211 @@
+			|| characterLowerPointOfCollision <= lowerBorder) {
+			
+			this.hit();
+		}
+	}
+	
+	private boolean checkCollision(int x, int y, int height, int width) {
+		int characterY = this.getPosition().getY();
+		int characterPointOfCollision = this.getPosition().getX() + (int) this.skin.getImage().getWidth(null);
+		int obstacleHeightLowerLimit = characterY + height;
+		int obstacleHeightUpperLimit = y;
+		int obstacleWidthLimit= x + width;
+		
+		//the purpose of this if is to check if the character is in the range of the obstacle
+		if (characterY <= obstacleHeightUpperLimit
+			&& characterY >= obstacleHeightLowerLimit) {
+					
+			//the purpose of this if is to check if the character hits the obstacle
+			if (characterPointOfCollision >= x
+				&& characterPointOfCollision <= obstacleWidthLimit) {
+								
+				return true;
+			}
+		}
+		
+		return false;
+	}
+	
+	
+	@Override
+	public void animate(Graphics2D canvas) {
+		
+		/*double angle = (((90 * (velocity + 20) / 20) - 90))*Math.PI/180;
+		
+		if(angle > Math.PI) {
+			angle = Math.PI;
+		}*/
+		
+		//this.rotateAndDrawImage(canvas, angle);
+		
+		//these variables will be changed with constants
+		int screenWidth = (int) Toolkit.getDefaultToolkit().getScreenSize().getWidth();
+		int screenHeight = (int) Toolkit.getDefaultToolkit().getScreenSize().getHeight();
+		
+		int x = this.getPosition().getX();
+		int y = this.getPosition().getY();
+		int width = (int) (screenWidth * 0.035);
+		int height = (int) (screenHeight * 0.045);
+		
+		
+		canvas.drawImage(this.getSkin().getImage(), x, y, width, height, null);
+		
+		if(this.isHit()) {
+			this.die();
+		}
+		
+		this.updatePosition(falling);
+	}
+	
+	/*private void rotateAndDrawImage(Graphics2D g2D, double angle) {
+		//these variables will be changed with constants
+		int screenWidth = (int) Toolkit.getDefaultToolkit().getScreenSize().getWidth();
+		int screenHeight = (int) Toolkit.getDefaultToolkit().getScreenSize().getHeight();
+		
+		int transformX = this.getPosition().getX() + this.getSkin().getImage().getWidth(null)/2;
+		int transformY = this.getPosition().getY() + this.getSkin().getImage().getHeight(null)/2;
+		int x = this.getPosition().getX();
+		int y = this.initialY + this.velocity;
+		int width = (int) (screenWidth * 0.035);
+		int height = (int) (screenHeight * 0.045);
+		
+		AffineTransform rotatedImage = AffineTransform.getRotateInstance(Math.toRadians(angle), transformX, transformY);
+		g2D.setTransform(rotatedImage);
+		g2D.drawImage(this.getSkin().getImage(), x, y, width, height, null);
+	}*/
+	
+	private void updatePosition(boolean falling) {
+		//these variable will be changed with constants
+		int screenHeight = (int) Toolkit.getDefaultToolkit().getScreenSize().getHeight();
+		
+		double value = falling ? screenHeight*0.0015 : -screenHeight*0.0015;
+		
+		this.getPosition().setX(this.getPosition().getX());
+		this.getPosition().setY((int) (this.getPosition().getY() + value));
+		
+		try {
+			Thread.sleep(100);

```

Comment:
Thread.sleep() stops the execution of the whole thing. A Timer should be used to time the change of direction (from going up to going down). A hint could be:
- a private field Timer timer;
- the initialization of the timer field in the constructor;
- a timer.start() call in the jump method that is performed only once in that jump (remember that the jump method is called many times per second, so you should ignore the timer.start call for the rest of the jump and start considering it again at the end of the jump, maybe using a boolean flag);
- an actionPerformed method (automatically generated by implementing ActionListener) that changes the direction of the movement (from up to down)

Reply:
Done